crates/aptos-crypto/README.md

```diff
@@ -4,32 +4,53 @@ title: Crypto
+  + Standardized in [RFC 5869](https://tools.ietf.org/html/rfc5869)
+  + Used to generate keys from a salt (optional), seed, and application-info (optional)
+- **Ed25519** signatures and (naive) multisignatures
+  + Based on the [ed25519-dalek](https://docs.rs/ed25519-dalek/) crate with additional security checks (e.g., for malleability)
+- **Boneh-Shacham-Lynn (BLS) multisignatures**
+  + Based on the [blst](https://docs.rs/blst/) crate
+  + Implemented on top of Barreto-Scott-Lynn BLS12-381 elliptic curves
+- The **[Noise Protocol Framework](http://www.noiseprotocol.org/)**
+  - Used to create authenticated and encrypted communications channels between validators
+- **X25519** key exchange
+  + Based on the [x25519-dalek](https://docs.rs/x25519-dalek) crate
+  + Used in our implementation of the [Noise Protocol Framework](http://www.noiseprotocol.org/)
+
+## Traits for safer cryptography implementation
+
+Before implementing a cryptographic primitive, be sure to read [`traits.rs`](src/traits.rs) and [`validatable.rs`](src/validatable.rs) to understand how to comply with our API as well as **some** of the security issues involved.
 
 ## How is this module organized?
 ```
     crypto/src
-    ├── hash.rs             # Hash function (SHA-3)
-    ├── hkdf.rs             # HKDF implementation (HMAC-based Extract-and-Expand Key Derivation Function based on RFC 5869)
-    ├── macros/             # Derivations for SilentDebug and SilentDisplay
-    ├── utils.rs            # Serialization utility functions
+    ├── bls12-381/          # BLS (multi)signatures over BLS12-381 curves
+    ├── unit_tests/         # Unit tests
     ├── lib.rs
     ├── ed25519.rs          # Ed25519 implementation of the signing/verification API in traits.rs
+    ├── hash.rs             # Hash function (SHA-3)
+    ├── hkdf.rs             # HKDF implementation
     ├── multi_ed25519.rs    # MultiEd25519 implementation of the signing/verification API in traits.rs
-    ├── x25519.rs           # X25519 wrapper
+    ├── noise.rs            # Noise Protocol Framework implementation
     ├── test_utils.rs
-    ├── traits.rs           # New API design and the necessary abstractions
-    └── unit_tests/         # Tests
+    ├── traits.rs           # Traits for safer implementations of signature schemes
+    ├── validatable.rs      # Traits for deferring validation of group elements (e.g., public keys, signatures)
+    └── x25519.rs           # X25519 implementation
+
 ```
 
-Note: This crate historically had support for BLS12381, ECVRF, and SlIP-0010, though were removed due to lack of use. The last git revision before there removal is 00301524.
+## Changelog
+
+ - This crate historically had support for (a different) BLS12-381, [EC-VRF](https://tools.ietf.org/id/draft-goldbe-vrf-01.html#rfc.section.5), and [SLIP-0010](https://github.com/satoshilabs/slips/blob/master/slip-0010.md), though were removed due to lack of use. The last git revision before the removal is 00301524.
```

Comment:
Let's delete this.

Reply:
Why? It would be a good reminder in case we need to take a look back.